Update shtok buttons prop file

ShtokyD opened the pull request:
1 and 2 buttons OS 5.x support for Battle Mode, new events ect.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -153,6 +240,69 @@ SaberShtokButtons() : PropBase() {}
 
+// Battle Mode
+
+#ifndef FETT263_BATTLE_MODE_ALWAYS_ON
```
You need to rename these defines.
Your prop file should be using SHTOK_* instead of FETT263_*
Also, it would be helpful to list them at the top of the file together with a description what they do.

ShtokyD replied:
I don't use these defines so I don't need them.

profezzorn replied:
If you don't want them, then you should delete them (and the code that is associated with it)

ShtokyD replied:
Deleted in this new version.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -153,6 +240,69 @@ SaberShtokButtons() : PropBase() {}
+      case EVENTID(BUTTON_NONE, EVENT_SWING, MODE_OFF | BUTTON_POWER):
+        if (!battle_mode_) {
+          battle_mode_ = true;
+	  if(millis() > 3000) { 
```
indentation looks wrong here and in other places, although that might because of github.
I can always fix it after it's checked in, but take a look.

ShtokyD replied:
I'm not sure about all this coding stuff, Fernando helped me to make this file and I tested it, it works fine :)

profezzorn replied:
I can fix the indentation, but the defines have to be fixed.

ShtokyD commented:
Added 0-button mode as 3rd mode and deleted fett263 defines. Can you please check if my prop file looks correct for 3 modes and for MOTION_TIMEOUT define that I then add in my config file to be able to turn the blade On after 5-10 minutes of no use by a swing or thrust motion.

ShtokyD commented:
Added     DetectShake();  and adjusted a few things.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -64,37 +177,170 @@ SaberShtokButtons() : PropBase() {}
+        }
+        break;
+
+// Lightning Block
```
This comment seems like it's not for anything.
Maybe remove it?

ShtokyD replied:
Deleted.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -64,37 +177,170 @@ SaberShtokButtons() : PropBase() {}
+// Lightning Block
+
+
+// Start or Stop Track
```
I think this comment can be removed too.

ShtokyD replied:
Deleted.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -1,47 +1,98 @@
+EFFECT(vmbegin); // for Begin Volume Menu
+EFFECT(vmend); // for End Volume Menu
+EFFECT(faston); // for EFFECT_FAST_ON
+EFFECT(blstbgn); // for Begin Multi-Blast
```
blstbgn and blstend doesn't seem to be used anywhere

ShtokyD replied:
that's for future changes, can it stay?

profezzorn replied:
no

ShtokyD replied:
Ok, will remove.

ShtokyD replied:
It's for Multi-Blast feature I believe and I use Multi-Blast.

profezzorn replied:
I can't find any references in this file to SFX_blstbgn or SFX_blstend which means they are not used and should be deleted.

ShtokyD replied:
Ok.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -1,47 +1,98 @@
+EFFECT(battery); // for EFFECT_BATTERY_LEVEL
+EFFECT(bmbegin); // for Begin Battle Mode
+EFFECT(bmend); // for End Battle Mode
+EFFECT(vmbegin); // for Begin Volume Menu
```
SFX_vmbegin and SFX_vmend doesn't seem to be used anywhere.

ShtokyD replied:
Deleted.

profezzorn replied:
Looks like they are still here to me.

ShtokyD replied:
Now should be deleted.

profezzorn commented on props/saber_shtok_buttons.h:
```diff
@@ -64,49 +217,185 @@ SaberShtokButtons() : PropBase() {}
+          return true;
+        }
+        break;
+
```
Why so many empty lines here?

ShtokyD replied:
to visually separate. Could use the array of ////// though.

profezzorn replied:
Might be better, not a big deal either way though.

profezzorn commented:
It looks like this CL now has a conflict, please resolve.

ShtokyD commented:
> 
> 
> It looks like this CL now has a conflict, please resolve.

I don't know what it means... all works fine on my sabers.
